implementations/rust/ockam/ockam_api/src/cloud/share/mod.rs

```diff
@@ -0,0 +1,102 @@
+use std::{fmt::Display, str::FromStr};
+
+use minicbor::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+mod accept;
+mod create;
+mod list;
+
+pub use accept::*;
+pub use create::*;
+pub use list::*;
+
+#[derive(Clone, Debug, PartialEq, Decode, Deserialize, Encode, Serialize)]
+#[cbor(index_only)]
+#[rustfmt::skip]
+pub enum RoleInShare {
+    #[n(0)] Admin,
+    #[n(1)] Guest,
+    #[n(2)] Service,
+}
+
+impl Display for RoleInShare {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        match self {
+            Self::Admin => write!(f, "admin"),
+            Self::Guest => write!(f, "guest"),
+            Self::Service => write!(f, "service_user"),
+        }
+    }
+}
+
+impl FromStr for RoleInShare {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "admin" => Ok(Self::Admin),
+            "guest" => Ok(Self::Guest),
+            "service" => Ok(Self::Service),
```

Comment:
The from/to conversions of the `Service` differ, should it be `service` in both cases?

Reply:
Thank you for catching this! I wrote those across two separate working sessions, but they should be in agreement, so I'll switch the From to also be `service_user`.

Reply:
This should be corrected in the last push.